apps/src/templates/instructions/InlineAudio.jsx

```diff
@@ -210,6 +237,55 @@ class InlineAudio extends React.Component {
     });
   }
 
+  getAutoplayTriggerElement() {
+    const {autoplayTriggerElementId} = this.props;
+    return autoplayTriggerElementId
+      ? document.getElementById(autoplayTriggerElementId)
+      : document;
+  }
+
+  // adds event listeners from the dom which trigger audio
+  // when a significant enough user has happened
+  addAudioAutoplayTrigger() {
+    AUDIO_ENABLING_DOM_EVENTS.forEach(event => {
+      const listeningElement = this.getAutoplayTriggerElement();
+      listeningElement.addEventListener(event, this.triggerAudio);
+    });
+  }
+
+  removeAudioAutoplayTrigger() {
+    AUDIO_ENABLING_DOM_EVENTS.forEach(event => {
+      const listeningElement = this.getAutoplayTriggerElement();
+      listeningElement.removeEventListener(event, this.triggerAudio);
+    });
+  }
+
+  playAudio() {
+    return this.getAudioElement()
+      .play()
+      .then(() => {
+        this.setState({playing: true});
+        this.recordPlayEvent();
+      })
+      .catch(err => {
+        const shouldAutoPlay =
+          this.props.ttsAutoplayEnabled && !this.state.autoplayed;
+
+        // there wasn't significant enough user interaction to play audio automatically
+        // for more information about this issue on Chrome, see
+        // https://developers.google.com/web/updates/2017/09/autoplay-policy-changes
+        if (err instanceof DOMException && shouldAutoPlay) {
+          this.addAudioAutoplayTrigger();
+        } else {
+          throw err;
+        }
+      });
+  }
+
+  triggerAudio() {
```

Comment:
NIT: imo `autoplayAudio` would be a more consistent name for this. the rest of the naming suggests we have an `autoplay` action that we potentially want to trigger by an event -- i.e. a trigger event causes an `autoplayAudio` action, not a `triggerAudio` action.